File: ADA_LabProgrammes/JavaProg/SelectionSort.java


public class SelectionSort {
    public static void main(String[] args) {
        int arr[] = {64, 25, 12, 22, 11};
        int min = 0;
        int temp = 0,i;
        for ( i = 0; i<arr.length-1; i++){
            min = i;
            for (int j = i+1; j<arr.length; j++){
                    if(arr[min]> arr[j]){
                        min = j;
                    }
            }
        

        temp = arr[min];
        arr[min] = arr[i];
        arr[i] = temp;
    }
    for (i =0; i< arr.length; i++){
        System.out.println(arr[i]);   
    }
}
    
}

File: ADA_LabProgrammes/JavaProg/selectionsort.cpp
#include<iostream>
#include<cstdlib>
#include<time.h>
using namespace std;

void selectionSort(int a[], int n){
    for (int i = 0; i< n-1; i++){
        int min = i;
        for (int j = i+1; j<n; j++){
            
            if (a[min] > a[j]){
                min = j;
            }

        // int temp;
        // temp = a[i];
        // a[i] = a[min];
        // a[min] = temp;
            swap(a[i],a[min]);
        }
    }
}



int main(){
    int upperlimit = 500;
    int lowerlimit = 100;
    int n;
    cout<< "enter the number of elements :"<< endl;
    cin>> n;
    srand(time(0));
    int *arr = new int[n];
    for(int i =0; i<n; i++){
        arr[i] =(rand()% (upperlimit - lowerlimit));
        cout<< arr[i]<<endl;
    }
    clock_t start = clock();
    selectionSort(arr,n);
    clock_t stop = clock();
    for(int i =0; i<n; i++){
        //arr[i] =(rand());
        cout<< arr[i]<<endl;
    }
    double dif = (stop - start)/ (double)CLOCKS_PER_SEC;

    cout<< "Time taken to sort "<<n<<"elements : "<<dif<<endl;
}